modify and rename part of the eviction / finalizer implementation

Jutho opened the pull request:
@GunnarFarneback and @Krastanov:  I've modified part of the implementation and renamed `eviction_callback` to `finalizer`. Let me know what you think. If you agree, I will merge and bump the version

Jutho committed:
rename and modify part of the eviction implementation

Jutho committed:
update readme, setup github ci, bump minor version

Krastanov commented:
I do not see anything unreasonable in the changes. Thanks for taking care of this! I would love to update my dependencies to this upcoming version.

GunnarFarneback commented:
I have no attachment to the names or the implementation details. It's the functionality that counts. But I'll review this to see if I can spot any problems.

GunnarFarneback commented on src/LRUCache.jl:
```diff
@@ -183,57 +181,54 @@ function Base.resize!(lru::LRU{K, V}; maxsize::Integer = lru.maxsize) where {K,
-    while !isempty(evictions)
-        key, value = pop!(evictions)
-        callback(key, value)
+function _finalize_evicitions!(finalizer, evictions)
```
Evictions is misspelled in this function name.

Also the definition of `_no_callback` can be removed, but github won't let me comment on lines outside the immediate context of the diff.

Jutho replied:
Thanks for carefully checking this. That's what you get from doing "find and replace all" to quickly :-).

GunnarFarneback commented:
I assume these changes weren't enough to avoid allocations where no finalizer is used, other than in `delete!` and `pop!`.

Jutho commented:
Indeed, I also tried to get rid of other allocations, but in the end the only way I could see how to do it was by making the type of `finalizer` an explicit type parameter of `LRU`, and I assumed that would be a breaking change. So I did not want to do that.

I could see one other way, which is to have an empty `Vector{Tuple{K,V}}()` associated with every instance of `LRU`, i.e. have it as an extra field. If `finalizer === nothing`, one can just use this one (and still check for `finalizer === nothing` later on so as not to push anything to it, cause that would again be not thread safe). It did not seem very elegant, but it might be worth considering.

Jutho committed:
address comments

Jutho commented:
I addressed the comments, but did not go with my proposal of adding an empty `Vector{Tuple{K,V}}()` to the `LRU` structure. I think that in the typical use cases of an LRU cache, the extra cost of allocating an empty vector are negligible compared to the other operations, even for a single insertion etc.

Jutho commented:
If CI turns green and @GunnarFarneback gives green light, I will merge and register the new version.

Jutho commented:
v1.3.0 is scheduled to be merged and should land soon!
